src/planner/binder/statement/bind_vacuum.cpp

```diff
@@ -3,84 +3,85 @@
-		D_ASSERT(stmt.info->column_id_map.empty());
-		auto bound_table = Bind(*stmt.info->ref);
-		if (bound_table->type != TableReferenceType::BASE_TABLE) {
-			throw InvalidInputException("Can only vacuum/analyze base tables!");
-		}
-		auto ref = unique_ptr_cast<BoundTableRef, BoundBaseTableRef>(std::move(bound_table));
-		auto &table = ref->table;
-		stmt.info->table = &table;
+	vector<unique_ptr<Expression>> select_list;
+	auto &columns = info.columns;
+	if (columns.empty()) {
+		// Empty means ALL columns should be vacuumed/analyzed
+		auto &get = ref->get->Cast<LogicalGet>();
+		columns.insert(columns.end(), get.names.begin(), get.names.end());
+	}
 
-		auto &columns = stmt.info->columns;
-		vector<unique_ptr<Expression>> select_list;
-		if (columns.empty()) {
-			// Empty means ALL columns should be vacuumed/analyzed
-			auto &get = ref->get->Cast<LogicalGet>();
-			columns.insert(columns.end(), get.names.begin(), get.names.end());
+	case_insensitive_set_t column_name_set;
+	vector<string> non_generated_column_names;
+	for (auto &col_name : columns) {
+		if (column_name_set.count(col_name) > 0) {
+			throw BinderException("Vacuum the same column twice(same name in column name list)");
 		}
-
-		case_insensitive_set_t column_name_set;
-		vector<string> non_generated_column_names;
-		for (auto &col_name : columns) {
-			if (column_name_set.count(col_name) > 0) {
-				throw BinderException("Vacuum the same column twice(same name in column name list)");
-			}
-			column_name_set.insert(col_name);
-			if (!table.ColumnExists(col_name)) {
-				throw BinderException("Column with name \"%s\" does not exist", col_name);
-			}
-			auto &col = table.GetColumn(col_name);
-			// ignore generated column
-			if (col.Generated()) {
-				continue;
-			}
-			non_generated_column_names.push_back(col_name);
-			ColumnRefExpression colref(col_name, table.name);
-			auto result = bind_context.BindColumn(colref, 0);
-			if (result.HasError()) {
-				result.error.Throw();
-			}
-			select_list.push_back(std::move(result.expression));
+		column_name_set.insert(col_name);
+		if (!table.ColumnExists(col_name)) {
+			throw BinderException("Column with name \"%s\" does not exist", col_name);
 		}
-		stmt.info->columns = std::move(non_generated_column_names);
-		if (!select_list.empty()) {
-			auto table_scan = CreatePlan(*ref);
-			D_ASSERT(table_scan->type == LogicalOperatorType::LOGICAL_GET);
-
-			auto &get = table_scan->Cast<LogicalGet>();
+		auto &col = table.GetColumn(col_name);
+		// ignore generated column
+		if (col.Generated()) {
+			continue;
+		}
+		non_generated_column_names.push_back(col_name);
+		ColumnRefExpression colref(col_name, table.name);
+		auto result = bind_context.BindColumn(colref, 0);
+		if (result.HasError()) {
+			result.error.Throw();
+		}
+		select_list.push_back(std::move(result.expression));
+	}
+	info.columns = std::move(non_generated_column_names);
+	// Creating a table without any physical columns is not supported
```

Comment:
I removed the `else` here that was setting `has_table` to false when `select_list` was empty.
Because this can't happen.

This was preparing for a scenario where a table exists that has no physical columns, only consisting of generated columns, which can't happen because that's not supported.